catalog/ui/search-ui/standard/src/main/webapp/js/model/Query.js

```diff
@@ -73,6 +73,11 @@ define([
                 this.listenTo(wreqr.vent, 'search:clearfilters', this.clearFilters);
 
                 this.filters = new Filter.Collection();
+                var that = this;
+
+                this.filters.on('change:north change:south change:east change:west', function(model) {
```

Comment:
You should use .listenTo as opposed to .on. If you use .on you also need to use .off or you'll end up with a memory leak. If you use, .listenTo, it handles it for you when this model is destroyed.